lib/api/objectDelete.js

```diff
@@ -87,6 +87,20 @@ function objectDeleteInternal(authInfo, request, log, isExpiration, cb) {
                     // versioning has been configured
                     return next(null, bucketMD, objMD);
                 }
+
+                if (versioningCfg && versioningCfg.Status === 'Enabled' &&
+                    objMD.versionId === reqVersionId && isExpiration) {
+                        log.warn('expiration is trying to delete a master version ' +
```

Comment:
should we filter out delete markers ?

Reply:
but if we are creating a delete marker, there is no `reqVersionId`, right? in this case, this is already handled

Reply:
Ah ok so you meant not logging when we delete an object that was already marked for deletion (i.e. objMD.isDeleteMarker is `true`). I added this check.